### ISSUE TITLE
quotes rule gets triggered when a backtick quoted string contains escaped newlines

### ISSUE THREAD

**Issue by remcohaszing** [eslint/eslint#9069]
**Tell us about your environment**

* **ESLint Version:** v4.3.0
* **Node Version:** v8.2.1
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
root: true

extends: airbnb
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
const helpMessage = `\
foo\
`;
```

**What did you expect to happen?**

I expected this to be ok, because this string contains escaped newlines.

**What actually happened? Please include the actual, raw output from ESLint.**

This triggers the `quotes` rule.

`eslint --fix` replaced it with:

```
const helpMessage = '\
foo\
';
```

This then triggers `no-multi-str`.

**Comment by not-an-aardvark** [comment]
I think this is working as intended, because there isn't any reason that the string needs to have backticks.. Actually, it seems like a bug to me that the original code didn't trigger `no-multi-str`.

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.

**Comment by avimar** [comment]
Seems to be my issues, too. I just upgraded to 4.5 from 3.8. Not sure why you're reporting this is intended.

ESLint Version: v4.5.0
Node Version: v6.11.2
npm Version: 5.3.0

```
    "env": {
        "es6": true,
        "node": true,
        "mocha": true
    },
"quotes": ["warn", "single", {"avoidEscape": true}]
```

**Code**:
```
const variable = `Some text which ends with a newline.\n\n`;
```

**What did you expect to happen?**
I expected this to be ok, because this string contains escaped newlines.
```
https://eslint.org/docs/rules/quotes
var backtick = `back\ntick`;  // backticks are allowed due to newline
```

What actually happened? Please include the actual, raw output from ESLint.
```
C:\Users\Avi\docker\bestfone\api\test.js
  1:18  warning  Strings must use singlequote                   quotes
```

**Comment by avimar** [comment]
Seems to be my issues, too. I just upgraded to 4.5 from 3.8. Not sure why you're reporting this is intended.

ESLint Version: v4.5.0
Node Version: v6.11.2
npm Version: 5.3.0

```
    "env": {
        "es6": true,
        "node": true,
        "mocha": true
    },
"quotes": ["warn", "single", {"avoidEscape": true}]
```

**Code**:
```
const variable = `Some text which ends with a newline.\n\n`;
```

**What did you expect to happen?**
I expected this to be ok, because this string contains escaped newlines.
```
https://eslint.org/docs/rules/quotes
var backtick = `back\ntick`;  // backticks are allowed due to newline
```

What actually happened? Please include the actual, raw output from ESLint.
```
C:\Users\Avi\docker\bestfone\api\test.js
  1:18  warning  Strings must use singlequote                   quotes
```

**Comment by not-an-aardvark** [comment]
I think the issue here is that the example with the newline in the documentation [here](https://eslint.org/docs/rules/quotes#double) is incorrect. The "correct" example should be:

```js
var backtick = `back
tick`;  // backticks are allowed due to newline
```

**Comment by avimar** [comment]
Oh. Doh. `\n` is totally allowed in single quoted text. It's just actually hitting enter ↵ that requires backticks.

Check my PR for documentation fix.
